sycl/source/detail/queue_impl.hpp

```diff
@@ -433,11 +438,6 @@ class queue_impl {
             ext::oneapi::cuda::property::queue::use_default_stream>()) {
       CreationFlags |= UR_QUEUE_FLAG_USE_DEFAULT_STREAM;
     }
-    if (PropList.has_property<ext::oneapi::property::queue::discard_events>()) {
-      // Pass this flag to the Level Zero adapter to be able to check it from
-      // queue property.
-      CreationFlags |= UR_QUEUE_FLAG_DISCARD_EVENTS;
-    }
```

Comment:
Besides things listed in the description this PR also seems to cleanup code for discarded events, please split unrelated changes to separate PRs.

Reply:
Yes, there are some changes around discard events, I hope this PR will get merged first: https://github.com/intel/llvm/pull/18059 and what's not covered there I can split to a separate PR.

Reply:
This has been merged as part of #18059